woocommerce/wc-admin
# Get post install scripts from gateway and enqueue in client

Fixes #6965 

* Adds a method to get post install scripts from payment gateways.
* Adds a reusable method to add scripts to the page via JS.
* Removes `useMemo` on inexpensive functions that wasn't working as expected (maybe these need to use `memo()`?)

Note that https://github.com/woocommerce/woocommerce-admin/pull/6918 will introduce a data store to get the payment gateway.  `@todo` added inline.

### Screenshots

<img width="695" alt="Screen Shot 2021-05-12 at 5 58 28 PM" src="https://user-images.githubusercontent.com/10561050/118049394-ac536e00-b34b-11eb-88ad-a654d9876013.png">


### Detailed test instructions:

1. Grab the latest version of [the remote tester](https://github.com/joshuatf/woocommerce-admin-remote-tester) and `npm run build`.
2. Install WooCommerce PayFast Gateway.
3. Inside the `class-wc-gateway-payfast.php` class add a getter for post install scripts.  Note that this just uses the remote tester's example slotfill.
```php
/**
 * Get the handles for post install scripts.
 */
public function get_post_install_script_handles() {
	if ( ! class_exists( 'RemotePaymentsTester' ) ) {
		return array();
	}

	RemotePaymentsTester::add_extension_register_script();
	return array( 'remote-payments-tester' );
}
```
4. Navigate to the PayFast payment task.
5. Note that the slotfill works after the install step.

Also note that this PR demonstrates some pre-existing wonkiness in how we setup these steps, unmount, and use possible issues with slotfill.  If we slotfill the entire step, there will most likely be a flash of the original stepper prior to replacement.  Some follow-ups will also be needed to clean up items around memoization (if necessary) and unmounting the original component.
